addons/web/static/src/js/fields/relational_fields.js

```diff
@@ -1157,6 +1178,21 @@ var FieldX2Many = AbstractField.extend(WidgetAdapterMixin, {
         }
         return this._super.apply(this, arguments);
     },
+    /**
+     * @override
+     * @param {Object} extraInfo
+     * @param {string} extraInfo.row
+     * @param {string} extraInfo.subFieldName
+     */
+    doQuickEdit: function (extraInfo) {
+        this._super(...arguments);
+
+        if (extraInfo.subFieldName) {
```

Comment:
I'm a bit concerned. In FieldBoolean, you check the mode in this method, but in Many2one and X2many you don't. I think it will crash if we get here in readonly (which I assume happens if the field is `readonly="1"`). If I'm right, this shows as I said above that the check should be in AbstractField